aws/protocol/query/middleware.go

```diff
@@ -0,0 +1,63 @@
+package query
```

Comment:
What was the thought around putting this middleware in the query protocol package?

Reply:
Primarily didn't have another place to put it, and the behavior was specific to the `query` protocol.